### DISCUSSION TITLE
Infrared Response from SSCHA?

### DISCUSSION THREAD

**Discussion by adenchfi** [SSCHAcode/python-sscha#46] [Q&A] [answered]
Hello,

I was wondering if you had any advice on how to extract infrared response from SSCHA, if possible. I guess my question is twofold - one, is there a tool for seeing if the SSCHA 'renormalized' modes are infrared active or not? And two, is there any way to account for matrix element effects? 

One idea is to simply include in the DFT calls, specify using my chosen DFT code (such as Quantum Espresso) that there be an electric field present, and I guess do the SSCHA again with an electric field present in all the DFT calls. Would that be conceptually ok with the SSCHA approach?

**Comment by mesonepigreco** [comment, marked as answer]
Hi, 
The easiest way is to perform a computation of the dielectric tensor and effective charges within a DFT code and copy them inside the first dynamical matrix file (as done in the usual output of quantum espresso).
Then you can use the quantum espresso dynmat.x code on the converged SSCHA dynamical matrix to get IR activity of the modes. 
The same utility is also implemented directly in CellConstructor:

```
import cellconstructor as CC, cellconstructor.Phonons

# Load the final SSCHA dynamical matrix (end of the SCHA minimization)
dyn = CC.Phonons.Phonons("dyn_sscha")

# Get the phonons at Gamma
w, pols = dyn.DyagDinQ(0) 

# Compute the IR intensities (the dynamical matrix must contain the effective charges)
I = dyn.GetIRIntensities()

# Print the result
print("Frequency [cm-1] | IR intensity")
print("\n".join(["{} {}".format(w[i] * CC.Units.RY_TO_CM, I[i]) for I in range(len(w))]))
```

However, these are IR modes of the auxiliary dynamical matrix of the SSCHA, which are very often good approximations of the true IR spectrum. The actual IR response, including phonon lifetime and eventual anharmonic satellite peaks, is obtained in the time-dependent SCHA as described [here](https://journals.aps.org/prb/abstract/10.1103/PhysRevB.103.104305) [(arxiv)](https://arxiv.org/abs/2011.14986v3)

This IR signal is a bit more complicated. You should compute the dynamical phonon Green function and contract it with the effective charge tensor. However, this is not implemented in the current version of the code. One needs to do it manually, as described in the linked paper.

However, the first method would do the job (it will give you the correct IR intensities, but be aware that the exact anharmonic frequency may be a little shifted in the presence of very strong anharmonicity)

**Reply by adenchfi** [reply]
Thanks very much! I'll take a better look at the linked paper, it's very interesting that the time dependent case has the same computational cost as the static case! Will update with questions if I have them.
